src/nsrt_learning/strips_learning/gen_to_spec_learner.py

```diff
@@ -377,7 +391,24 @@ def _finalize_pnad_side_predicates(pnad: PartialNSRTAndDatastore) -> None:
                 side_predicates.add(atom.predicate)
             for atom in next_atoms - segment.final_atoms:
                 side_predicates.add(atom.predicate)
-        pnad.op = pnad.op.copy_with(side_predicates=side_predicates)
```

Comment:
if this ever makes it into master, we should rename this function to something like `_finalize_pnad_side_predicates_and_keep_effects`